test/e2e/framework/util.go

```diff
@@ -1356,16 +1358,33 @@ func RandomSuffix() string {
 
 // ExpectEqual expects the specified two are the same, otherwise an exception raises
 func ExpectEqual(actual interface{}, extra interface{}, explain ...interface{}) {
+	if reflect.TypeOf(actual) != reflect.TypeOf(extra) {
+		e2elog.Logf("Unexpected unequal occurred: %v and %v", actual, extra)
+		debug.PrintStack()
+	} else if reflect.ValueOf(actual) != reflect.ValueOf(extra) {
```

Comment:
This is not the type of comparison you want. From the godocs:
```
To compare two Values, compare the results of the Interface method. Using == on two Values does not compare the underlying values they represent.
```

I wouldn't even use the `Interface` since it gets confusing to anyone about when/how they can use certain `reflect` methods.

It seems like in these two cases we should be using the actual match that ginkgo will do, just leverage those without the `Expect`:

```
if isEqual, _ := gomega.Equal(extra).Match(actual); isEqual{
...
}
```

Double-check me on that syntax but I think thats the way it would be written.

Reply:
Hi @johnSchnake , based on the doc https://onsi.github.io/gomega/#provided-matchers, I can't find a use like `gomega.Equal(extra).Match(actual)` but I think I can use `reflect.DeepEqual` to compare ACTUAL with EXPECTED, which is the `gomega.Equal` implementation.

Reply:
They are all exported types/interfaces so we should just leverage what they've already written. For instance, they dont just do `reflect.DeepEqual` but have an optimization in place. I don't want to have to maintain a separate method which could drift and doesn't have any justification/benefit over using theirs.

```
//For details on writing custom matchers, check out: http://onsi.github.io/gomega/#adding-your-own-matchers
type GomegaMatcher interface {
    Match(actual interface{}) (success bool, err error)
    FailureMessage(actual interface{}) (message string)
...
```

Equal matcher implementation:
https://github.com/kubernetes/kubernetes/blob/master/vendor/github.com/onsi/gomega/matchers/equal_matcher.go#L15-L28

Reply:
@johnSchnake could you re-review? Thanks!